resources/services/network/interfaces.go

```diff
@@ -0,0 +1,405 @@
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.PrivateIPAddressVersion"),
+					},
+					{
+						Name:        "private_ip_allocation_method",
+						Description: "Private IP address allocation method.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.PrivateIPAllocationMethod"),
+					},
+					{
+						Name:        "private_link_connection_properties",
+						Description: "PrivateLinkConnection properties for the network interface.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.PrivateLinkConnectionProperties"),
+					},
+					{
+						Name:        "provisioning_state",
+						Description: "READ-ONLY; The provisioning state of the network interface IP configuration.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.ProvisioningState"),
+					},
+					{
+						Name:        "public_ip_address",
+						Description: "Public IP address bound to the IP configuration.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.PublicIPAddress"),
+					},
+					{
+						Name:        "subnet_id",
+						Description: "subnet ID of network interface ip configuration",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.Subnet.ID"),
+					},
+					{
+						Name:        "virtual_network_taps",
+						Description: "The reference to Virtual Network Taps.",
+						Type:        schema.TypeJSON,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.VirtualNetworkTaps"),
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchNetworkInterfaces(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
+	svc := meta.(*client.Client).Services().Network.Interfaces
+	response, err := svc.ListAll(ctx)
+	if err != nil {
+		return err
+	}
+	for response.NotDone() {
+		res <- response.Values()
+		if err := response.NextWithContext(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func fetchNetworkInterfaceIPConfigurations(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	ni, ok := parent.Item.(network.Interface)
+	if !ok {
+		return fmt.Errorf("expected to have network.Interface but got %T", parent.Item)
+	}
+	res <- *ni.IPConfigurations
+	return nil
```

Comment:
`nil pointer exception` possibility, check other similar places
```suggestion
        if ni.IPConfigurations != nil
	    res <- *ni.IPConfigurations
	}
	return nil
```